### PR TITLE
Dispatched key events have a correct type (closes #941)

### LINKED ISSUE #941: [todomvc] Wrong dispatched key event types
It does not pass check `instanseof`
It leads problem in testcafe https://github.com/DevExpress/testcafe/issues/964

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @churkin @AlexanderMoskovkin

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
TestCafe tests shouldn't work with these changes (reauire fix in testcafe)

**Comment by helen-dikareva** [comment]
FPR

File: test/client/fixtures/sandbox/event/event-simulator-test.js [hunk 1]
```diff
@@ -471,3 +482,15 @@ asyncTest('should not define window.event property if event is raised in iframe
     document.body.appendChild(iframe);
 });
 
+test('wrong type of key event (GH-941)', function () {
```
Comment (MarinaRukavitsyna) [inline test/client/fixtures/sandbox/event/event-simulator-test.js]: test('wrong type of the key event (GH-941)'

File: test/client/fixtures/sandbox/event/event-simulator-test.js [hunk 2]
```diff
@@ -445,6 +455,7 @@ asyncTest('hammerhead functions should not be in strict mode (GH-344)', function
 
     eventSimulator.click(button[0]);
 });
+
 asyncTest('should not define window.event property if event is raised in iframe for element of top window', function () {
```
Comment (MarinaRukavitsyna) [inline test/client/fixtures/sandbox/event/event-simulator-test.js]: asyncTest('should not define the window.event property if the event is raised in iframe for the element of top window'

File: src/client/sandbox/event/simulator.js [hunk 3]
```diff
@@ -316,24 +327,84 @@ export default class EventSimulator {
...
+
+            Object.defineProperty(ev, 'which', {
+                get: () => args.which
             });
 
-            return this._raiseDispatchEvent(el, ev, args);
+            var prevented   = false;
+            var returnValue = true;
+
+            // NOTE: dispatchEvent method does not return false in case of preventDefault was
```
Comment (MarinaRukavitsyna) [inline src/client/sandbox/event/simulator.js]: // NOTE: the dispatchEvent method does not return false in the case when preventDefault was 
// raised for events that were created with the KeyboardEvent constructor

File: src/client/sandbox/event/simulator.js [hunk 4]
```diff
@@ -316,24 +327,84 @@ export default class EventSimulator {
...
+            });
+        }
+        else if (nativeMethods.documentCreateEvent) {
+            ev = nativeMethods.documentCreateEvent.call(document, 'KeyboardEvent');
+
+            ev.initKeyboardEvent(args.type, args.canBubble, args.cancelable, args.view, '', 0, EventSimulator._getModifiersAsString(args), false, '');
+        }
+
+        if (ev) {
+            // NOTE: keyCode, charCode and which properties are not assigned after creation of KeyboardEvent
```
Comment (MarinaRukavitsyna) [inline src/client/sandbox/event/simulator.js]: // NOTE: the window.event.keyCode, window.event.charCode and window.event.which properties are not assigned after KeyboardEvent is created

File: src/client/sandbox/event/simulator.js [hunk 5]
```diff
@@ -316,24 +327,84 @@ export default class EventSimulator {
...
+                        prevented = true;
+
+                        return false;
+                    },
+
+                    set: () => void 0
+                });
+            }
+
+            // NOTE: dispatchEvent method does not return false in case of returnValue was assigned to false (only in MSEdge)
```
Comment (MarinaRukavitsyna) [inline src/client/sandbox/event/simulator.js]: // NOTE: the dispatchEvent method does not return false when returnValue was set to false (only in MSEdge)

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Pull request merged**